platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/crypto/CryptoArgsProvider.java

```diff
@@ -51,22 +53,26 @@ public class CryptoArgsProvider {
      */
     static Stream<Arguments> basicTestArgs() throws Exception {
         Instant start = Instant.now();
-        final AddressBookAndCerts addressBookAndCerts = loadAddressBookWithKeys(NUMBER_OF_ADDRESSES);
+        final RosterAndCerts rosterAndCerts = loadAddressBookWithKeys(NUMBER_OF_ADDRESSES);
         start = Instant.now();
         final AddressBook genAB = createAddressBook(NUMBER_OF_ADDRESSES);
         final Map<NodeId, KeysAndCerts> genC = CryptoStatic.generateKeysAndCerts(genAB);
         start = Instant.now();
         return Stream.of(
-                Arguments.of(addressBookAndCerts.addressBook(), addressBookAndCerts.nodeIdKeysAndCertsMap()),
-                Arguments.of(genAB, genC));
+                Arguments.of(rosterAndCerts.roster(), rosterAndCerts.nodeIdKeysAndCertsMap()),
+                Arguments.of(RosterRetriever.buildRoster(genAB), genC));
     }
 
     public static AddressBook createAddressBook(final int size) {
-        final AddressBook addresses = RandomAddressBookBuilder.create(Randotron.create())
+        final Roster roster = RandomRosterBuilder.create(Randotron.create())
                 .withSize(size)
                 .withWeightDistributionStrategy(WeightDistributionStrategy.BALANCED)
                 .build();
 
+        // We still use the keys injection mechanism from the EnhancedKeyStoreLoader and CryptoStatic,
+        // so we use an AddressBook here.
+        AddressBook addresses = RosterUtils.buildAddressBook(roster);
```

Comment:
I've created an issue to bookmark replacing the use of `AddressBook` in the crypto loading classes: https://github.com/hashgraph/hedera-services/issues/16402

Reply:
Thanks.